airflow/contrib/operators/cloudml_operator.py

```diff
@@ -14,16 +14,321 @@
+                project_id, region, data_format, input_paths, output_path,
+                model_name, version_name, uri, max_worker_count,
+                runtime_version)
+        except ValueError as e:
+            logging.error(
+                'Cannot create batch prediction job request due to: {}'
+                .format(str(e)))
+            raise
+
+        self.prediction_job_request = {
+            'jobId': job_id,
+            'predictionInput': prediction_input
+        }
+
+    def execute(self, context):
+        hook = CloudMLHook(self.gcp_conn_id, self.delegate_to)
+        job_id = _normalize_cloudml_job_id(
+            self.prediction_job_request['jobId'])
+        try:
+            existing_job = hook.get_job(self.project_id, job_id)
+            logging.info(
+                'Job with job_id {} already exist: {}.'.format(
+                    job_id,
+                    existing_job))
+            finished_prediction_job = hook.wait_for_job_done(
+                self.project_id,
+                job_id)
+        except errors.HttpError as e:
+            if e.resp.status == 404:
+                logging.error(
+                    'Job with job_id {} does not exist. Will create it.'
+                    .format(job_id))
+                finished_prediction_job = hook.create_job(
+                    self.project_id,
+                    self.prediction_job_request)
+            else:
+                raise
+
+        if finished_prediction_job['state'] != 'SUCCEEDED':
+            logging.error(
+                'Batch prediction job failed: %s',
+                str(finished_prediction_job))
+            raise RuntimeError(finished_prediction_job['errorMessage'])
+
+        return finished_prediction_job['predictionOutput']
+
+
+class CloudMLModelOperator(BaseOperator):
+    """
+    Operator for managing a Google Cloud ML model.
+
+    :param model: A dictionary containing the information about the model.
+        If the `operation` is `create`, then the `model` parameter should
+        contain all the information about this model such as `name`.
+
+        If the `operation` is `get`, the `model` parameter
+        should contain the `name` of the model.
+    :type model: dict
+
+    :param project_name: The Google Cloud project name to which CloudML
+        model belongs.
+    :type project_name: string
+
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: string
+
+    :param operation: The operation to perform. Available operations are:
+        'create': Creates a new model as provided by the `model` parameter.
+        'get': Gets a particular model where the name is specified in `model`.
+
+    :param delegate_to: The account to impersonate, if any.
+        For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: string
+    """
+
+    template_fields = [
+        '_model',
```

Comment:
Thanks Chirs! the project_name is not intended for templating on purpose.  

To answer the question of the funky rebase, I changed some of the code related to PR2379 for styles only. The main code changes are on the CloudMLBatchOperator (PR2390).